graphql_compiler/query_formatting/cypher_formatting.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+import datetime
+import json
+from string import Template
+
+import arrow
+from graphql import GraphQLBoolean, GraphQLFloat, GraphQLID, GraphQLInt, GraphQLList, GraphQLString
+import six
+
+from ..compiler import CYPHER_LANGUAGE
+from ..compiler.helpers import strip_non_null_from_type
+from ..exceptions import GraphQLInvalidArgumentError
+from ..schema import GraphQLDate, GraphQLDateTime, GraphQLDecimal
+from .representations import represent_float_as_str, type_check_and_str
+
+
+def _safe_cypher_string(argument_value):
+    """Sanitize and represent a string argument in Cypher."""
+    if not isinstance(argument_value, six.string_types):
+        if isinstance(argument_value, bytes):  # likely to only happen in py2
+            argument_value = argument_value.decode('utf-8')
+        else:
+            raise GraphQLInvalidArgumentError(u'Attempting to convert a non-string into a string: '
+                                              u'{}'.format(argument_value))
+
+    # Using JSON encoding means that all unicode literals and special chars
+    # (e.g. newlines and backslashes) are replaced by appropriate escape sequences.
+    # Unlike with Gremlin, unescaped dollar signs $ are not a problem when contained in a
+    # string literal in Cypher because they do not allow for arbitrary code execution.
+    escaped_and_quoted = json.dumps(argument_value)
+    return escaped_and_quoted
+
+
+def _safe_cypher_decimal(argument_value):
+    """Cypher doesn't support decimals, only ints and floats, so we'll raise an error here."""
+    raise NotImplementedError(u'Cypher doesn\'t support Decimals, only ints and floats. See '
+                              u'OpenCypher documentation. Argument value was {}'.
```

Comment:
nit: I'd move the `.` to the next line, otherwise the syntax highlighting gets a bit confused and thinks that `format()` is a top-level function rather than a method on the string itself.